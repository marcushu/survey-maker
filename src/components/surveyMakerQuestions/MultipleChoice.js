import React, { useState } from 'react';
import PropTypes from 'prop-types';
import './questionInsert.css';

export default function MultipleChoice({ addQuestion }) {
  const [question, setquestion] = useState("");
  const [multChoiceOptions, setmultChoiceOptions] = useState("");

  const handleAddQuestion = () => {
    if (!!question && !!multChoiceOptions) {
      const cleanList = multChoiceOptions
        .split(',')
        .map(s => s.trim().replace(' ', '_'));

      addQuestion({
        questionType: "MULTCHOICE",
        question: question,
        choices: cleanList
      });

      setquestion("");
    } else {
      alert("Please enter question and choices");
    }

  }

  return (
    <span>
      <input type="text"
        className="inputText"
        placeholder="Your question here..."
        value={question}
        onChange={e => setquestion(e.target.value)}>
      </input>
      &nbsp;
      <input type="text"
        className="inputText"
        style={{ width: "150px" }}
        placeholder="ok, good, great..."
        value={multChoiceOptions}
        onChange={e => setmultChoiceOptions(e.target.value)}>
      </input>
      &nbsp;
      <button className="add" onClick={handleAddQuestion}>+</button>
    </span>
  )
}

MultipleChoice.propTypes = {
  addQuestion: PropTypes.func
}