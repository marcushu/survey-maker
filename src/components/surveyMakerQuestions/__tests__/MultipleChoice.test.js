import React from 'react';
import {  fireEvent, render } from '@testing-library/react';
import MultipleChoice from '../MultipleChoice';

describe("multiple choice",() => {

    const mockfunc = jest.fn();

    it("renders", () => {
        const { asFragment } = render(<MultipleChoice addQuestion={mockfunc}/>);

        expect(asFragment()).toMatchSnapshot();
    });


    it("correctly returns a list of options", () => {
        const { getByPlaceholderText, getByRole } = render(<MultipleChoice addQuestion={mockfunc}/>);

        fireEvent.change(getByPlaceholderText("Your question here..."),
            { target: { value: "any text will do"}});

        fireEvent.change(getByPlaceholderText("ok, good, great..."), 
            { target: { value: ['one', 'two', 'three'] } });
        
        fireEvent.click(getByRole("button"));

        expect(mockfunc).toHaveBeenCalledTimes(1);
        expect(mockfunc.mock.calls[0][0].choices).toStrictEqual([ 'one', 'two', 'three' ]);
    })
})