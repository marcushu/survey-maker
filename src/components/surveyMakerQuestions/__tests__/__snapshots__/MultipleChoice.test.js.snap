// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`multiple choice renders 1`] = `
<DocumentFragment>
  <span>
    <input
      class="inputText"
      placeholder="Your question here..."
      type="text"
      value=""
    />
     
    <input
      class="inputText"
      placeholder="ok, good, great..."
      style="width: 150px;"
      type="text"
      value=""
    />
     
    <button
      class="add"
    >
      +
    </button>
  </span>
</DocumentFragment>
`;
